x-pack/spec/modules/azure/azure_module_config_generator_spec.rb

```diff
@@ -0,0 +1,146 @@
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License;
+# you may not use this file except in compliance with the Elastic License.
+
+require 'x-pack/logstash_registry'
+require 'logstash-core'
+require 'logstash/settings'
+require 'logstash/util/modules_setting_array'
+require 'logstash/modules/scaffold'
+require 'azure_module_config_generator'
+
+describe LogStash::Azure::ConfigGenerator do
+
+  # always clone and work with the clone since settings are global and can bleed between tests
+  let(:settings) {settings = LogStash::SETTINGS.clone}
+
+  describe "valid configuration" do
+    it "advanced" do
+      settings.from_yaml(::File.join(File.dirname(__FILE__), "samples", "yaml"), "logstash.advanced.yml")
```

Comment:
suggestion: I like to move as much as possible the variables to `let`s, so the code in the `it` block focuses on the test itself instead of the setup.
Also simplifies reuse when doing multiple tests in the same describe/context block.

Reply:
would add additional `describe`s (like would needed below) to push the setup into 'let's ?  resulting in a describe/let/it block per test ?

Reply:
I think so, probably something like:


```ruby
describe LogStash::Azure::ConfigGenerator do
  let(:settings) { settings.from_yaml(::File.join(File.dirname(__FILE__), "samples", "yaml"), config_path) }
  let(:module_config) do
    module_hash = settings.get("modules").find {|m| m["name"] == 'azure'} }
    scaffolding = LogStash::Modules::Scaffold.new('azure', ::File.join(File.dirname(__FILE__), '..', '..', '..', 'modules', 'azure', 'configuration')) }
    LogStash::Modules::LogStashConfig.new(scaffolding, module_hash)
  end 
  describe "basic mode" do
    context "with a valid configuration" do
      let(:config_path) { "path to valid config" }
      it "transpiles correctly" do
      end
    end
    context "when 'event_hub_connections' isn't set" do
       let(:config_path) { "path to config without event_hub_connections" }
       it "raises an error" do
       end
    end
  end
  describe "advanced mode" do
    # same structure as above
  end
end
```